### ISSUE TITLE
Create streamlit page for "local file" library uploads

### ISSUE THREAD

**Issue by fvankrieken** [NYCPlanning/data-engineering#1017]
Some of our library datasets that end up in `edm-recipes` come from a manual upload - someone has emailed us a csv, we've grabbed something via an ftp on NYCNET, etc. Their templates look roughly like [this](https://github.com/NYCPlanning/data-engineering/blob/main/dcpy/library/templates/dcp_pops.yml#L6) for source. For these, we
- download the file from wherever it comes from
- place the file in the correct location on our computer and/or edit path in template yml
- run library locally to archive

This isn't TOO painful, but for a lot of datasets it certainly is tedious to make sure these paths align when something a little more drag-and-drop would make this process much simpler.

This is why a basic streamlit page would be very helpful, basically with
- text input (or selectbox input) to enter dataset name
- text input to enter version
- file upload dialog

We'd then want to
- archive this raw file to `edm-recipes/inbox/{dataset}/{version}/{filename}` so that we have the raw file archived
- call library for the given dataset, but pointing to this file in `inbox` for the `path`. This is probably the trickiest part, because we'll need to figure out the simplest way to feed this into `library` code which is a little opaque at times

So to summarize what needs to be done
- [x] add new page "ingest dataset" to apps/qa
- [x] add inputs for this new page
- [x] write helper function within QA app to upload file to `edm-recipes/inbox/{dataset}/{version}/{filename}`
- [ ] write helper function within QA app to read in `template` file (using `library` utilities) but then replace the `path` with the new s3 path pushed to in the last step
- [ ] call appropriate `library` function using this new "config" object

**Comment by fvankrieken** [comment]
This can be tested running streamlit locally (`python3 -m streamlit run src/index.py` from within `apps/qa`). Streamlit is nice in that you don't need to restart the server to see code changes - as you develop, the app detects changes and offers to reload to reflect your changes
